src/load/azext_load/data_plane/utils/utils.py

```diff
@@ -340,12 +344,66 @@ def convert_yaml_to_test(cmd, data):
     if data.get(LoadTestConfigKeys.AUTOSTOP) is not None:
         new_body["autoStopCriteria"] = utils_yaml_config.yaml_parse_autostop_criteria(data=data)
 
-    utils_yaml_config.update_engine_reference_identity(new_body, data)
+    utils_yaml_config.update_reference_identities(new_body, data)
     logger.debug("Converted yaml to test body: %s", new_body)
     return new_body
 # pylint: enable=line-too-long
 
 
+# pylint: disable=line-too-long
+# Disabling this because if conditions are too long
+def parse_app_comps_and_server_metrics(data):
+    app_components_yaml = data.get(LoadTestConfigKeys.APP_COMPONENTS)
+    app_components = {}
+    server_metrics = {}
+    add_defaults_to_app_copmponents = dict()
+    if app_components_yaml is None:
+        return None, None, None
+    if not isinstance(app_components_yaml, list):
+        raise InvalidArgumentValueError("App component name should be of type list")
+    for app_component in app_components_yaml:
+        resource_id = app_component.get(LoadTestConfigKeys.RESOURCEID)
```

Comment:
This check should be after dict check otherwise it will error out?

Reply:
didnot get this exact comment, i think resource_id is kept before the dict check, and i moved it dowen now.